project.json for .NET core

caleblloyd opened the pull request:
Use this to build and publish for .NET Framework >= 3.5 and .NET Core

You'll net to install the .NET core SDK: https://www.microsoft.com/net/core

Once you have that installed, navigate to the repository's HtmlDiff folder, then run:

    dotnet restore
    dotnet build
    dotnet pack
    dotnet publish

caleblloyd commented on HtmlDiff/Utils.cs:
```diff
@@ -9,7 +9,7 @@ public static class Utils
         private static Regex tagWordRegex = new Regex(@"<[^\s>]+", RegexOptions.Compiled);
         private static Regex whitespaceRegex = new Regex("^(\\s|&nbsp;)+$", RegexOptions.Compiled);
-        private static Regex wordRegex = new Regex(@"[\w\#@]+", RegexOptions.Compiled | RegexOptions.ECMAScript);
+        private static Regex wordRegex = new Regex(@"[\w\#@]+", RegexOptions.ECMAScript);
```
.NET Core has a bug and throws a runtime error with these lines OR'd.  Had to remove RegexOptions.Compiled for the time being

Rohland replied:
@caleblloyd To be safe, it might be worth look at using a conditional directive [as outlined here](https://docs.microsoft.com/en-us/dotnet/articles/core/tutorials/libraries) for the .NET Core version until the bug is resolved.

brianpos commented:
Have you done any performance testing at scale with this?
Seems like without the compiled portion, speed/memory could become an issue (don't know though)

caleblloyd commented:
@brianpos I am on linux so only running .NET Core, therefore can't benchmark it on any other version of the framework on a with the compiled option.  With this change in .NET core I am diffing two 5000 line HTML files in approximately 1.5 seconds on an Intel i5-6200u machine.

My best guess is that it will have little impact on performance.  This is a very small regex, and I doubt the time to run it through the JIT compiler is much slower than the natively compiled implementation.  Of course the best way to tell would be running it on a different version of the framework with and without the compiled option.

I've got an open issue to CoreFX https://github.com/dotnet/corefx/issues/11270

jasonwilliams200OK commented:
There are instructions on how to conduct performance tracing on Linux https://github.com/dotnet/coreclr/blob/master/Documentation/project-docs/linux-performance-tracing.md.

caleblloyd commented on HtmlDiff/Utils.cs:
```diff
@@ -9,7 +9,14 @@ public static class Utils
+        /// bug in .NET core prevents RegexOptions.Compiled | RegexOptions.ECMAScript
+        /// tracked at: https://github.com/dotnet/corefx/issues/11270
+#if NETSTANDARD1_3
+        private static Regex wordRegex = new Regex(@"[\w\#@]+", RegexOptions.ECMAScript);
```
Added conditional compilation to regex

caleblloyd commented:
@Rohland good idea on the conditional compilation flag, I've added that

I've also changed the project.json to target netstandard1.3 so that this can be used as a library in any version of the framework that supports netstandard1.3 or higher

"dotnet publish -c Release" is no longer necessary, "dotnet pack -c Release" will pack this as a library for upload to nuget

Lastly, I've updated the tests project to run in .NET Core from the test directory using:

    dotnet restore
    dotnet test

caleblloyd commented:
@Rohland any chance of this getting in so the .NET core folks can use?  I'm happy to pack and ship it to nuget if you want to give me access (nuget username: caleblloyd)
